programs/zk-token-proof/src/lib.rs

```diff
@@ -132,6 +132,11 @@ declare_process_instruction!(process_instruction, 0, |invoke_context| {
 
     match instruction {
         ProofInstruction::CloseContextState => {
+            if native_programs_consume_cu {
```

Comment:
Why do we need the `if native_programs_consume_cu {` check at all? Can't we just always consume the CUs?

Reply:
Yes, we don't need this check and can just remove it. It was added in long time ago and it was never really removed. The newer instructions that were added in don't have this feature gate check.

If I remove it here, then some of the CI tests fail due to some proof instructions exceeding compute unit limit. So I thought I would just add the compute units for close instruction in this PR to get that backported quickly and then remove the feature gate in a separate PR. As long as the proof program activation is coordinated with https://github.com/solana-labs/solana/issues/30620, then I thought this would be okay.